UnitTests/GravatarTests/AvatarImageNameProviderTests.cs

```diff
@@ -1,13 +1,15 @@
-﻿using FluentAssertions;
+﻿using System;
+using FluentAssertions;
 using Gravatar;
+using NSubstitute.ExceptionExtensions;
 using NUnit.Framework;
 
 namespace GravatarTests
 {
     [TestFixture]
     public class AvatarImageNameProviderTests
     {
-        private AvatarImageNameProvider _provider;
+        private IImageNameProvider _provider;
```

Comment:
❕ Previous code was correct. This is a test class for a concrete type, not for an interface. (Also noted in a couple other places in the tests.)

Reply:
I'm indifferent here. The class only exposes non-constructor members via the interface.

Reply:
The concrete type is being instantiated. It's just being tested via its interface. Given its only non-private members are the constructor and interface members, this is fine. This technique, combined with coverage metrics, enforces that consumers use the interface, and that the interfaces are adequately tested.